hls-plugin-api/src/Ide/Plugin/ConfigUtils.hs

```diff
@@ -26,9 +26,9 @@ import           Language.LSP.Types
 pluginsToDefaultConfig :: IdePlugins a -> A.Value
 pluginsToDefaultConfig IdePlugins {..} =
   A.Object $
-    HMap.adjust
+    adjustJson
```

Comment:
This is the only one I can't think of a nice way around :(

Reply:
damn, i was hoping we can take ride of the Compat module. Would you keep it only for adjust or do you think is better to copy here the cpp?

Reply:
Okay, so what we'd probably want to avoid this is a aeson-lens library that handled both versions and gave us lenses for adjusting inside objects. 

Indeed, it looks like https://hackage.haskell.org/package/lens-aeson-1.1.3/docs/Data-Aeson-Lens.html#v:_Object would do it, and is set up to work with both pre and post 2.0 aeson. We could pull that in if we wanted, it's not got any deps we don't have already, I think?

Reply:
There's also an `Ixed Value` instance.

Reply:
Afaics we are already depending on all dependencies of lens-aeson

Reply:
> There's also an `Ixed Value` instance.

Sorry but how can we leverage that instance?

Reply:
I'd need to go and poke at all the lens stuff to figure it out, but I *think* we could use something like https://hackage.haskell.org/package/lens-4.17/docs/Control-Lens-At.html#v:iix. I can maybe try and take a look later.

Reply:
Many thanks intermmediate/advanced lenses usage is still hard to me 😅

Reply:
Or maybe just using the prism to focus on the object case and then using `at`? I always have to work this stuff out from scratch every time :sweat_smile: